src/schematools/contrib/django/factories.py

```diff
@@ -131,17 +131,40 @@ def _get_related_name(self):
         If the linked table describes the other end of the relationship,
         this field will also be included in the model.
         """
-        if self.field._parent_table.has_parent_table:
+        parent_table = self.field.table
+        if parent_table.is_nested_table:
             # Won't ever show related name for internal tables
-            return to_snake_case(self.field._parent_table["originalID"])
-        elif self.field._parent_table.is_through_table:
-            # Need this for walking over the through table for the "_links" section.
-            return to_snake_case(f"{self.field._parent_table.id}_through_{self.field.id}")
+            return to_snake_case(parent_table["originalID"])
+        elif parent_table.is_through_table:
+            # This provides a reverse-link from each FK in the M2M table to the linked tables,
+            # allowing to walk *inside* the M2M table instead of over it.
+            # For debugging purposes these names are clarified depending on what direction
+            # the key takes.
+            m2m_field = parent_table.parent_table_field
+            through_fields = parent_table["throughFields"]
+            if self.field.name == through_fields[0]:
+                # First model, can resemble the original field name,
+                return f"rev_m2m_{to_snake_case(m2m_field.name)}"
+            elif self.field.name == through_fields[1]:
+                # Second model, can resemble the reverse name if it exists.
+                m2m_reverse_name = m2m_field.reverse_relation
+                if m2m_reverse_name is not None:
+                    # As the field exists on the second model,
+                    # let the reverse relation also reflect that.
+                    return f"rev_m2m_{to_snake_case(m2m_reverse_name.id)}"
+
+            # By default, create something unique and recognizable.
+            # The "m2m" would be snake_cased as m_2_m, so it's added afterwards.
+            # The parent table ID already has the main field name included, but for tables
+            # with a self-reference, the field is still added to guarantee uniqueness.
+            return "rev_m2m_" + to_snake_case(f"{parent_table.id}_via_{self.field.id}")
```

Comment:
Shouldn't this use `self.field.name` instead of the `id`?

Reply:
nope :-) The `.name` is used for the `shortname` in SQL. All Django model field names use the long identifiers, so this reverse name is kept in line with that to clearly see which field it's opposite end is.